alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/SteeringActionImpl.kt

```diff
@@ -1,63 +1,60 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
 package it.unibo.alchemist.model.implementations.actions
 
-import it.unibo.alchemist.model.implementations.positions.AbstractEuclideanPosition
-import it.unibo.alchemist.model.implementations.routes.PolygonalChain
 import it.unibo.alchemist.model.interfaces.Environment
-import it.unibo.alchemist.model.interfaces.Node
 import it.unibo.alchemist.model.interfaces.Pedestrian
+import it.unibo.alchemist.model.interfaces.Position
 import it.unibo.alchemist.model.interfaces.Reaction
 import it.unibo.alchemist.model.interfaces.SteeringAction
-import it.unibo.alchemist.model.interfaces.movestrategies.RoutingStrategy
-import it.unibo.alchemist.model.interfaces.movestrategies.SpeedSelectionStrategy
-import it.unibo.alchemist.model.interfaces.movestrategies.TargetSelectionStrategy
+import it.unibo.alchemist.model.interfaces.geometry.Vector
 
 /**
- * Generic implementation of an action adhering the SteeringAction interface.
- *
- * @param env
- *          the environment inside which the pedestrian moves.
- * @param pedestrian
- *          the owner of this action.
- * @param target
- *          the strategy used to compute the next target.
- * @param speed
- *          the speed selection strategy.
- * @param routing
- *          the routing strategy.
+ * An abstract [SteeringAction], deriving from [AbstractMoveNode].
  */
-open class SteeringActionImpl<T, P : AbstractEuclideanPosition<P>> @JvmOverloads constructor(
-    private val env: Environment<T, P>,
-    reaction: Reaction<T>,
-    pedestrian: Pedestrian<T>,
-    private val target: TargetSelectionStrategy<P>,
-    private val speed: SpeedSelectionStrategy<P> = SpeedSelectionStrategy { pedestrian.speed() / reaction.rate },
-    private val routing: RoutingStrategy<P> = RoutingStrategy { p1, p2 -> PolygonalChain(p1, p2) }
-) : AbstractConfigurableMoveNode<T, P>(
-    env,
-    pedestrian,
-    routing,
-    target,
-    speed
-), SteeringAction<T, P> {
-
-    override fun cloneAction(n: Node<T>, r: Reaction<T>) =
-        SteeringActionImpl(env, r, n as Pedestrian<T>, target, speed, routing)
+abstract class SteeringActionImpl<T, P>(
```

Comment:
Refactor to `AbstractSteeringAction`. Avoid `Impl` when possible, and make non-instantiability clear from name.